core/src/main/java/io/grpc/Deadline.java

```diff
@@ -139,4 +139,9 @@ public long timeRemaining(TimeUnit unit) {
     Preconditions.checkNotNull(scheduler, "scheduler");
     return scheduler.schedule(task, deadlineNanos - System.nanoTime(), TimeUnit.NANOSECONDS);
   }
+
+  @Override
+  public String toString() {
+    return String.valueOf(deadlineNanos);
```

Comment:
I wouldn't mind if it was:
```java
return "" + timeRemaining(NANOSECONDS) + "ns from now";
```

GrpcUtil is another option. I think it would probably be more useful to be relative instead of printing out the date/time, but date/time would be more helpful than just the raw deadlineNanos.